src/Hamiltonians/HubbardRealSpace.jl

```diff
@@ -113,22 +113,63 @@ It is implemented recursively to ensure type stability.
     return self + row + _interactions(as, rest)
 end
 
+"""
+    trap_potential(add::SingleComponentFockAddress, pot)
+
+Calculate trap potential energy of a single-component address `add`. The (precomputed) 
+potential energy per particle at each point in the lattice is given by the vector `pot`.
+```math
+V = \\sum_{d=1}^D \\sum_{i=1}^M V_d x_{i,d}^2 n_i,
+```
+where ``x = (x_1, ..., x_D)`` is the distance of site `i` to the centre of the trap and 
+`n_i` is the number of particles at site `i`. `M` is the total number of sites on the 
+lattice.
+"""
+function trap_potential(add::SingleComponentFockAddress, pot)
+    pe = 0.0
+    for (n,i) in occupied_modes(add)
+        pe += n * pot[i]
+    end
+    return pe
+end
+
+"""
+    trap_potential(add::CompositeFS, pot::Matrix)
+
+Calculate potential energy of a multicomponent address as the sum of the potential energy
+of each component. Each column of `pot` is a precomputed vector of potential energy for 
+that component at each lattice point.
+"""
+function trap_potential(add::CompositeFS, pot::Matrix)
```

Comment:
Further to my comment above, it might be sufficient to document the function `trap_potential` with a single docstring rather than documenting the different methods (but I'm fine with either).